pkg/reconciler/topic/topic.go

```diff
@@ -0,0 +1,420 @@
+		// cache may be stale and we don't want to overwrite a prior update
+		// to status with this stale state.
+
+	} else if _, uErr := c.updateStatus(ctx, Topic); uErr != nil {
+		logger.Warnw("Failed to update Topic status", zap.Error(uErr))
+		c.Recorder.Eventf(Topic, corev1.EventTypeWarning, "UpdateFailed",
+			"Failed to update status for Topic %q: %v", Topic.Name, uErr)
+		return uErr
+	} else if reconcileErr == nil {
+		// There was a difference and updateStatus did not return an error.
+		c.Recorder.Eventf(Topic, corev1.EventTypeNormal, "Updated", "Updated Topic %q", Topic.GetName())
+	}
+	if reconcileErr != nil {
+		c.Recorder.Event(Topic, corev1.EventTypeWarning, "InternalError", reconcileErr.Error())
+	}
+	return reconcileErr
+}
+
+func (c *Reconciler) reconcile(ctx context.Context, topic *v1alpha1.Topic) error {
+	logger := logging.FromContext(ctx)
+
+	topic.Status.InitializeConditions()
+
+	if topic.GetDeletionTimestamp() != nil {
+		logger.Debug("Topic Deleting.", zap.Any("propagationPolicy", topic.Spec.PropagationPolicy))
+
+		switch topic.Spec.PropagationPolicy {
+		case v1alpha1.TopicPolicyCreateDelete:
+			// Ensure the Topic is deleted.
+			state, err := c.EnsureTopicDeleted(ctx, topic, topic.Spec.Project, topic.Status.TopicID)
+			switch state {
+			case pubsub.OpsJobGetFailed:
+				logger.Error("Failed to get Topic ops job.", zap.Any("state", state), zap.Error(err))
+				return err
+
+			case pubsub.OpsJobCreated:
+				// If we created a job to make a subscription, then add the finalizer and update the status.
+				topic.Status.MarkTopicOperating(
+					"Deleting",
+					"Created Job to delete topic %q.",
+					topic.Status.TopicID)
+				return nil
+
+			case pubsub.OpsJobCompleteSuccessful:
+				topic.Status.MarkNoTopic("Deleted", "Successfully deleted topic %q.", topic.Status.TopicID)
+				topic.Status.TopicID = ""
+				removeFinalizer(topic)
+
+			case pubsub.OpsJobCreateFailed, pubsub.OpsJobCompleteFailed:
+				logger.Error("Failed to delete topic.", zap.Any("state", state), zap.Error(err))
+
+				msg := "unknown"
+				if err != nil {
+					msg = err.Error()
+				}
+				topic.Status.MarkNoTopic(
+					"DeleteFailed",
+					"Failed to delete topic: %q.",
+					msg)
+				return err
+			}
+
+		default:
+			removeFinalizer(topic)
+		}
+
+		return nil
+	}
+
+	// Set the topic being used.
+	topic.Status.TopicID = topic.Spec.Topic
+
+	switch topic.Spec.PropagationPolicy {
+	case v1alpha1.TopicPolicyCreateDelete, v1alpha1.TopicPolicyCreateNoDelete:
+		state, err := c.EnsureTopicCreated(ctx, topic, topic.Spec.Project, topic.Status.TopicID)
+		// Check state.
+		switch state {
+		case pubsub.OpsJobGetFailed:
+			logger.Error("Failed to get topic ops job.",
+				zap.Any("propagationPolicy", topic.Spec.PropagationPolicy),
+				zap.Any("state", state),
+				zap.Error(err))
+			return err
+
+		case pubsub.OpsJobCreated:
+			// If we created a job to make a subscription, then add the finalizer and update the status.
```

Comment:
Seems like this comment should be `// If we created a job to make a topic`

Reply:
oops. copy/pasta